src/main/java/com/tmh/controller/admin/AdminController.java

```diff
@@ -9,30 +9,37 @@
 
 import com.tmh.service.CategoryService;
 import com.tmh.service.OrderService;
+import com.tmh.service.ProductService;
 import com.tmh.service.UserService;
 
 @Controller
 @RequestMapping(value = "/admin")
 public class AdminController {
-	
+
 	private static final Logger logger = Logger.getLogger(AdminController.class);
-	
+
 	@Autowired
 	protected CategoryService categoryService;
-	
+
 	@Autowired
 	protected UserService userService;
-	
+
 	@Autowired
-	protected OrderService orderService;
+	protected ProductService productService;
 
 	@Autowired
 	protected MessageSource messageSource;
 	
+	@Autowired
+	protected OrderService orderService;
+
+
 	protected void loadModelAttribute(Model model) {
 		model.addAttribute("categories", categoryService.findAll());
 		model.addAttribute("users", userService.findAll());
-		model.addAttribute("orders", orderService.findAll());
+		model.addAttribute("categories_menu", categoryService.findAll());
+		model.addAttribute("products", productService.findAll());
```

Comment:
Chú load quá nhìu đối tượng mà ko dùng hết thì rất tốn hiệu suất nhé
trừ khi object đó, dùng cho toàn bộ view của toàn bộ controller
còn ko, hãy chỉ load nó ở những method của controller cần dùng nó

Reply:
Dạ bọn em định làm cái menu đa cấp ở layout mà có dữ liệu động ạ, ví dụ như là thêm hay xóa category thì nó cũng tự động cập nhật trên menu nên bọn em dùng cái group mapping này để load...

Reply:
còn nếu để load riêng từng cái thì bọn em viết thêm hàm mới cũng dc chị ạ

Reply:
Để bọn em tính lại rồi sửa ạ , huhu